src/modules/sih/sih.hpp

```diff
@@ -170,11 +177,23 @@ class Sih : public ModuleBase<Sih>, public ModuleParams, public px4::ScheduledWo
 	void publish_sih();
 	void generate_fw_aerodynamics();
 	void generate_ts_aerodynamics();
+	void sensor_step();
+
+#if defined(ENABLE_LOCKSTEP_SCHEDULER)
+	void lockstep_loop();
+	int _lockstep_component{-1};
+	uint64_t _current_simulation_time_us{0};
+	uint64_t _last_iteration_wall_time_us{0};
+#endif
 
-	perf_counter_t  _loop_perf{perf_alloc(PC_ELAPSED, MODULE_NAME": cycle")};
-	perf_counter_t  _loop_interval_perf{perf_alloc(PC_INTERVAL, MODULE_NAME": cycle interval")};
```

Comment:
I'd like to keep the perf, why did you remove them?

Reply:
They got lost when moving the module back from the work queue. Re-added them!

Reply:
All the perfs are showing 0 (not only the sih module), when using the lockstep.

When building with `_nolockstep`,  sih cycle interval shows 0
`sih: cycle interval: 0 events, 0.00us avg, min 0us max 0us 0.000us rms`

Reply:
Oh right, the interval perf count needs to be in the actual loop, fixed that.
The perf counters showing 0 in lockstep is to be expected, since all time measurements from perf are based on simulation time. Simulation time only advamces at each loop iteration in the sih module in lockstep, so all time elapsed will be 0. Also, it will show you 4ms iteration time for the loop (250Hz) no matter how much you speed up simulation, since that is constant in simulation time.